pkg/generate/code/resource_reference_test.go

```diff
@@ -31,13 +31,22 @@ func Test_ReferenceFieldsValidation_NoReferenceConfig(t *testing.T) {
 			GeneratorConfigFile: "generator-with-reference.yaml",
 		})
 
-	crd := testutil.GetCRDByName(t, g, "Api")
+	crd := testutil.GetCRDByName(t, g, "Integration")
 	require.NotNil(crd)
-	expected := ""
-	assert.Equal(expected, code.ReferenceFieldsValidation(crd, "ko", 1))
+
+	field := crd.Fields["APIID"]
+	expected :=
+		`	if ko.Spec.APIRef != nil && ko.Spec.APIID != nil {
+		return ackerr.ResourceReferenceAndIDNotSupportedFor("APIID", "APIRef")
+	}
+	if ko.Spec.APIRef == nil && ko.Spec.APIID == nil {
+		return ackerr.ResourceReferenceOrIDRequiredFor("APIID", "APIRef")
+	}
```

Comment:
One of the good use case of https://github.com/aws-controllers-k8s/pkg/pull/14/files#diff-4b1976072e607d66abb6e181c6699c9be57d46ada282e77ee704c6cdbebb17e7R36-R48 i believe. I know it's a pain to check for the zero values as well. But users could provide `apiID: ""` ...

Reply:
`lo` also supports [`IsEmpty`](https://github.com/samber/lo#isempty)